### ISSUE TITLE
SuppressionXpathSingleFilter suppresses all violations when used with the default configuration

### ISSUE THREAD

**Issue by Vyom-Yadav** [checkstyle/checkstyle#11460]
I have read check documentation: https://checkstyle.sourceforge.io/config_filters.html#SuppressionXpathSingleFilter
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="SuppressionXpathSingleFilter"/>
    <module name="MissingJavadocTypeCheck">
    </module>
  </module>
</module>

/var/tmp $ cat Test.java
public class Test { // expected violation as all properties of filter are null.
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.0-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```
---

**Describe what you expect in detail.**

The default configuration has all the properties set to null but still the violation is suppressed, if I remove or comment the filter then a violation is thrown-
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
<!--    <module name="SuppressionXpathSingleFilter"/>-->
    <module name="MissingJavadocTypeCheck">
    </module>
  </module>
</module>
```
```
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.0-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /var/tmp/Test.java:1:1: Missing a Javadoc comment. [MissingJavadocType]
Audit done.
Checkstyle ends with 1 errors.
```

**Comment by pbludov** [comment]
This case is strange. `SuppressionXpathSingleFilter` with all properties set to `null' is useless.
It is better to throw a configuration error than silently do nothing if the filter is misconfigured.

**Comment by pbludov** [comment]
At least, this behaviour should be documented, similar to [this](https://checkstyle.sourceforge.io/config_coding.html#MatchXpath).

**Comment by romani** [comment]
>It is better to throw a configuration error than silently do nothing if the filter is misconfigured.

There are a lot of modules that do nothing useful or match nothing by default.

**Comment by strkkk** [comment]
Fix is merged
